### PR TITLE
feat: add reveal with flyout to bookmarkListMenu

### PR DESCRIPTION
**Pull request by shawncothran**
For: https://github.com/lonelyplanet/dotcom-profile/issues/312

### CODE REVIEW

File: src/components/bookmarkListMenu/index.jsx [hunk 1]
```diff
@@ -144,13 +159,11 @@ class BookmarkListMenu extends React.Component {
 BookmarkListMenu.propTypes = {
   innerRef: PropTypes.func.isRequired,
   children: PropTypes.element.isRequired,
-  iconName: PropTypes.oneOf([
-    "Ellipsis",
-    "Share",
-  ]).isRequired,
+  iconName: PropTypes.string.isRequired,
```
Comment (thomasthesecond) [inline src/components/bookmarkListMenu/index.jsx]: Can we keep `oneOf` here? I don't think any of our icons should be used here.

File: package.json [hunk 2]
```diff
@@ -76,10 +80,6 @@
     "react": "^15.0.0 || ^16.0.0"
   },
   "devDependencies": {
-    "@storybook/addon-actions": "^3.1.9",
-    "@storybook/addon-knobs": "^3.0.0",
-    "@storybook/addon-links": "^3.0.0",
-    "@storybook/react": "^3.0.1",
```
Comment (thomasthesecond) [inline package.json]: Storybook docs recommend these to be a devDependency

File: src/components/bookmarkListMenu/index.jsx [hunk 3]
```diff
@@ -104,15 +110,24 @@ class BookmarkListMenu extends React.Component {
...
-          iconName={iconName}
-          label={iconLabel}
-          size={40}
-          owns={optionsId}
-          color={colors.textPrimary}
-          backgroundColor={colors.bgPrimary}
-        />
+        {reveal ?
+          <IconRevealButton
+            id={"poiId"}
```
Comment (thomasthesecond) [inline src/components/bookmarkListMenu/index.jsx]: Can this be generated dynamically based on the `id` prop?

Comment (thomasthesecond) [reply src/components/bookmarkListMenu/index.jsx]: ```id={`${id}-IconRevealButton`}```

File: src/components/bookmarkListMenu/index.jsx [hunk 4]
```diff
@@ -144,13 +159,11 @@ class BookmarkListMenu extends React.Component {
...
   children: PropTypes.element.isRequired,
-  iconName: PropTypes.oneOf([
-    "Ellipsis",
-    "Share",
-  ]).isRequired,
+  iconName: PropTypes.string.isRequired,
   iconLabel: PropTypes.string.isRequired,
   id: PropTypes.string,
   className: PropTypes.string,
+  reveal: PropTypes.bool,
```
Comment (thomasthesecond) [inline src/components/bookmarkListMenu/index.jsx]: Can we add a default prop as false for `reveal`?

File: stories/index.jsx [hunk 5]
```diff
@@ -484,8 +482,9 @@ storiesOf("Bookmark list menu", module)
   .add("Default", () => (
     <Center>
       <BookmarkListMenu
-        iconName={select("Icon name", ["Ellipsis", "Share"], "Ellipsis")}
+        iconName={text("Icon name", "Ellipsis")}
```
Comment (thomasthesecond) [inline stories/index.jsx]: This would be reverted based on my comment above.

File: src/components/bookmarkListMenu/index.jsx [hunk 6]
```diff
@@ -2,13 +2,18 @@ import React from "react";
...
 import Flyout from "../flyout";
 import IconButton from "../iconButton";
+import IconRevealButton from "../iconRevealButton";
 import colors from "../../styles/colors";
 import timing from "../../styles/timing";
+import iconFromString from "../../utils/icon";
 import propTypes from "../../utils/propTypes";
 import clickOutside from "../../hoc/clickOutside";
 
+const _ = { upperFirst };
```
Comment (thomasthesecond) [inline src/components/bookmarkListMenu/index.jsx]: May not need this if `iconName` prop type is reverted to `oneOf`

File: package.json (hunk 2 above)
Comment (shawncothran) [reply package.json]: Oh ok, I changed them based on eslint

File: src/components/bookmarkListMenu/index.jsx (hunk 3 above)
Comment (shawncothran) [reply src/components/bookmarkListMenu/index.jsx]: oversight

**Pull request merged**